protocol/contracts/governance/ElderCandidate.sol

```diff
@@ -0,0 +1,163 @@
+
+    /**
+     * @dev Change the nomination fee
+     * 
+     * Designed to be low, but a defense in depth to stop peeps putting up a nomination
+     * for the lolz.
+     */
+    function setNominationFee(uint256 _nominationFee) external onlyOwner {
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @dev periodically update the merkle tree root to match discord membership.
+     * NOTE: If a user has already minted their 'elder candidate' NFT, this doesn't
+     * remove it for them.
+     */
+    //TODO(butlerji): Should we make this 'manager' - as I can imagine an endstate where this is
+    // auto-updated from the dao games backend server.
+    function updateMerkleRoot(bytes32 _merkleRoot) external onlyOwner {
+        merkleRoot = _merkleRoot;
+    }
+
+    // TODO(butlerji): Confirm if merkleProof is 0x0, no candidate can be added
+    // TODO(butlerji): Confirm a fee of 0 works
+    // TODO(butlerji>nicho): The voting UI should only shows candidates in the intersection of the set
+    //                       ElderCandidate NFT and active Discord Users (post MVP/to be discussed)
+    /**
+     * Enable the given discord ID as a candidate.
+     */
+    // NOTE(butlerji): For MVP, we'll create a small merkle tree with *just* the people who requested nomination on discord
+    //                 We can generalise later with an merkle root for everyone in discord + a nominal fee to stop spamming
+    //                 nominations for the lolz
+    function nominate(
+        address account,
+        uint256 discordId,
+        bytes32[] calldata merkleProof
+    ) external {
+        require(account != address(0), "MasterCandidate: Address cannot be 0x0");
+        require(MerkleProof.verify(merkleProof, merkleRoot, keccak256(abi.encode(account, discordId.toString()))), "MasterCandidate: candidate not in temple discord");
+
+        SafeERC20.safeTransferFrom(nominationFeeToken, account, address(this), nominationFee);
+        _safeMint(account, discordId);
+    }
+
+    function toggleEndorsement(uint256 discordId) external {
+        _toggleEndorsement(msg.sender, discordId);
+    }
+
+    function toggleEndorsements(uint256[] memory discordIds) external {
+        _toggleEndorsements(msg.sender, discordIds);
+    }
+
+    /**
+     * @notice Withdraw for another user (gasless for the vault token holder)
+     * (assuming the owner has given authority for the caller to act on their behalf)
+     *
+     * @dev amount is explicit, to allow use case of partial vault withdrawals
+     */
+    // TODO(butlerji): Need to test this works. The idea is I'd sign a message on mainnet, which gets sent to our backend, then sent to 
+    //                 the mempool on whatever L2 we are running this on.
+    function toggleEndorsementsFor(address account, uint256[] memory discordIds, uint256 deadline, uint8 v, bytes32 r, bytes32 s) public {
+        require(block.timestamp <= deadline, "MasterCandidate: expired deadline");
```

Comment:
After jamming:

The idea is that the backend off-chain server relayer would be a synchronous API that the dapp calls.

To protect from voter spamming (eg create XX many accounts which can all vote and clog up the contract & count), it would check that the account has a certain number of temple. eg `$TEMPLE.balanceOf(voter) > 10`

So:
1. user votes by dapp - this signs a tx and calls the server side off-chain relayer API
2. Server side relayer verifies the signer is the account, and then gets the balance of that account. Returns error if under a threshold
3. Server side relayer executes the tx on the L2 (can perhaps simulate to make sure it won't fail, to save gas)
4. Returns success to Dapp only when the tx has completed mining, or error upon failure.

So I guess it'll need to be a thread pool type setup of API requests.